src/System.Collections/tests/Performance/Perf.HashSet.cs

```diff
@@ -0,0 +1,501 @@
+using System.Text;
+using Xunit;
+
+/// <summary>
+/// Performs quick perf test of HashSet operations. Tests with T = int
+/// 
+/// This is intended for the regular perf runs to catch perf regressions. 
+/// Use HashSetDetailedTests for detailed perf analysis.
+/// </summary>
+public class Perf_HashSet
+{
+    private const int InitialSetSize = 8000000;
+    private const int InitialSetSize_small = 320000;
+    private const int MaxStartSize = 32000;
+
+    [Benchmark]
+    [InlineData(InitialSetSize, 1)]
+    [InlineData(InitialSetSize, 100)]
+    [InlineData(InitialSetSize, 10000)]
+    public static void Add(int initialSetSize, int countToAdd)
+    {
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            RandomTGenerator<int> intGenerator = new RandomTGenerator<int>(InstanceCreators.IntGenerator);
+            // n is varying start size of set
+            int[] startingElements = intGenerator.MakeNewTs(initialSetSize);
+            int[] stuffToAdd = intGenerator.MakeNewTs(countToAdd);
+            HashSet<int> theSet = new HashSet<int>();
+            theSet.UnionWith(startingElements);
+
+            using (iteration.StartMeasurement())
+            {
+                foreach (int thing in stuffToAdd)
+                {
+                    theSet.Add(thing);
+                }
+            }
+        }
+    }
+
+    [Benchmark]
+    [InlineData(InitialSetSize, 1)]
+    [InlineData(InitialSetSize, 100)]
+    [InlineData(InitialSetSize, 10000)]
+    public static void Contains(int initialSetSize, int countToCheck)
+    {
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            RandomTGenerator<int> intGenerator = new RandomTGenerator<int>(InstanceCreators.IntGenerator);
+            int[] startingElements = intGenerator.MakeNewTs(initialSetSize);
+            int[] subsetToCheck = intGenerator.GenerateSelectionSubset(startingElements, countToCheck);
+            bool present;
+            HashSet<int> theSet = new HashSet<int>();
+            theSet.UnionWith(startingElements);
+
+            using (iteration.StartMeasurement())
+            {
+                foreach (int thing in subsetToCheck)
+                {
+                    present = theSet.Contains(thing);
+                }
+            }
+        }
+    }
+
+    [Benchmark]
+    [InlineData(InitialSetSize, 1)]
+    [InlineData(InitialSetSize, 100)]
+    [InlineData(InitialSetSize, 10000)]
+    public static void Remove(int initialSetSize, int countToRemove)
+    {
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            RandomTGenerator<int> intGenerator = new RandomTGenerator<int>(InstanceCreators.IntGenerator);
+            int[] startingElements = intGenerator.MakeNewTs(initialSetSize);
+            int[] stuffToRemove = intGenerator.GenerateSelectionSubset(startingElements, countToRemove);
+            HashSet<int> theSet = new HashSet<int>();
+            theSet.UnionWith(startingElements);
```

Comment:
HashSet has a ctor that takes an enumerable.  Any reason not to use it rather than having a second line that uses UnionWith?